### ISSUE TITLE
Collection fields are unnecessarily initialized to "null"

### ISSUE THREAD

**Issue by saberya** [joelittlejohn/jsonschema2pojo#1346]
GenerationConfig has a option of "isInitializeCollections",
if it is true,
the result is 
```
private List<Foo> fooList = new ArrayList<>();
```
if it is false
the result is
```
private List<Foo> fooList = null;
```
How can i get the result
```
private List<Foo> fooList;

```

**Comment by unkish** [comment]
Hi

`private List<Foo> fooList;` is equally same as `private List<Foo> fooList = null;`

Please see [JLS 4.12.5. Initial Values of Variables ](https://docs.oracle.com/javase/specs/jls/se7/html/jls-4.html#jls-4.12.5)

> For all reference types (§4.3), the default value is null.

**Comment by saberya** [comment]
@unkish 
but 
```
private List<Foo> fooList = null;
```
is not what i expect.
Beacuse the java Class with this kind of expression is not a good habit to definition a field.

**Comment by unkish** [comment]
I'm sorry but I don't understand why is that a problem or what kind of problem are you trying to solve

**Comment by unkish** [comment]
At present it's not possible to get the described result without either:

- making changes to generator
- overriding `DefaultRule` with custom one

**Comment by joelittlejohn** [comment]
Not sure when this began happening, but I agree it would be nice to remove it. It's annoying noise.

Feel free to submit a PR for this.

**Comment by joelittlejohn** [comment]
Closed by #1347
